packages/camera/camera/lib/src/camera_preview.dart

```diff
@@ -44,38 +44,46 @@ class CameraPreview extends StatelessWidget {
   }
 
   Widget _wrapInRotatedBox({required Widget child}) {
-    if (kIsWeb || defaultTargetPlatform != TargetPlatform.android) {
-      return child;
+    if (defaultTargetPlatform == TargetPlatform.android) {
+      if (kIsWeb) {
```

Comment:
Doesn't this only check if the app was compiled for web?: https://api.flutter.dev/flutter/foundation/kIsWeb-constant.html

This should cause this to always be false if `defaultTargetPlatform == TargetPlatform.android` is true.

Reply:
I think this misunderstanding is likely a large part of the issue here. The assumption appears to be that there exists only iOS, Android, (Desktop) web, when in fact there are iOS, Android, iOS web, Android web, desktop web, etc. See the [documentation](https://api.flutter.dev/flutter/foundation/defaultTargetPlatform.html). The commit message for this change says "[camera] Only native Android needs to have its preview rotated." That was my assumption.